api/parts/data/fetch.js

```diff
@@ -1827,4 +1829,94 @@ function union(x, y) {
     return res;
 }
 
+/**
+* Get data for jobs listing for jobs api
+* @param {string} metric - name of the collection where to get data from
+* @param {params} params - params object with app_id and date
+*/
+fetch.fetchJobs = async function(metric, params) {
+    if (params.qstring.name) {
+        fetch.jobDetails(metric, params);
+    }
+    fetch.alljobs(metric, params);
+};
+
+/**
+* Get all jobs grouped by job name for jobs api
+* @param {string} metric - name of the collection where to get data from
+* @param {params} params - params object with app_id and date
+*/
+fetch.alljobs = async function(metric, params) {
+    const columns = ["name", "schedule", "next", "finished", "status", "total"];
+    let sort = {};
+    let total = await countlyDb._native.collection('jobs').aggregate([
+        {
+            $group: { _id: "$name" }
+        },
+        {
+            $count: 'total'
+        }
+    ]).toArray();
+    total = total.length > 0 ? total[0].total : 0;
+    const pipeline = [
+        {
+            $sort: {
+                finished: -1
+            }
+        },
+        {
+            $group: {
+                _id: "$name",
+                name: {$first: "$name"},
+                status: {$first: "$status"},
+                schedule: {$first: "$schedule"},
+                next: {$first: "$next"},
+                finished: {$first: "$finished"},
+                total: {$sum: 1}
+            }
+        }
+    ];
+    if (params.qstring.sSearch) {
+        let namesMatch = await countlyDb._native.collection('jobs').distinct('name', {name: {$regex: new RegExp(params.qstring.sSearch, "i")}});
```

Comment:
Codacy found an issue: [Found non-literal argument to RegExp Constructor](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5834029)